vm-setup/roles/v1aX_integration_test/tasks/node_reuse.yml

```diff
@@ -0,0 +1,328 @@
+    shell: |
+        kubectl get bmh -n "{{ NAMESPACE }}" | grep -w ready | awk '{print $1}'
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    register: ready_bmh
+    retries: 200
+    delay: 2
+    until: deprovisioning_bmh_name.stdout in ready_bmh.stdout_lines
+
+  - name: Check if just deprovisioned and became ready BMH is re-used for next provisioning.
+    shell: |
+        kubectl get bmh -n "{{ NAMESPACE }}" | grep -w provisioning | awk '{print $1}'
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    register: provisioning_bmh
+    retries: 150
+    delay: 2
+    until: deprovisioning_bmh_name.stdout in provisioning_bmh.stdout_lines
+
+  - name: Wait until two machines become running and updated with new "{{ UPGRADED_K8S_VERSION }}" k8s version.
+    shell: |
+        kubectl get machines -n "{{ NAMESPACE }}" -o json | jq -r '.items[] | select (.status.phase == "Running") | select(.spec.version == "{{ UPGRADED_K8S_VERSION }}") | .status.phase' | grep -c "Running"
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    register: updated_machines_partly
+    retries: 200
+    delay: 20
+    ignore_errors: yes
+    until: updated_machines_partly.stdout|int > 1
+
+  - pause:
+      minutes: 5
+
+  - name: Untaint all CP nodes after upgrade of two controlplane nodes.
+    shell: |
+        kubectl taint nodes --all node-role.kubernetes.io/master-
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    ignore_errors: yes
+
+  - name: Wait until all "{{ NUMBER_OF_BMH_KCP }}" machines become running and updated with new "{{ UPGRADED_K8S_VERSION }}" k8s version.
+    shell: |
+        kubectl get machines -n "{{ NAMESPACE }}" -o json | jq -r '.items[] | select (.status.phase == "Running") | select(.spec.version == "{{ UPGRADED_K8S_VERSION }}") | .status.phase' | grep -c "Running"
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    register: updated_machines_all
+    retries: 200
+    delay: 20
+    ignore_errors: yes
+    until: updated_machines_all.stdout|int == 3
+
+  - name: Get provisioned BMH names and UUIDs mapping after upgrade in KubeadmControlPlane node reuse test scenario.
+    shell: |
+        kubectl get bmh -A -o json | jq '.items[]| select (.status.provisioning.state == "provisioned")
+        | "metal3/"+.metadata.name+"="+"metal3://"+.metadata.uid' | cut -f2 -d\" | sort > /tmp/after_upgrade_mapping.txt
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+
+  - name: Check diff of before and after upgrade mappings to make sure same BMHs' were reused in KubeadmControlPlane node reuse test scenario.
+    shell: |
+        diff /tmp/before_upgrade_mapping.txt /tmp/after_upgrade_mapping.txt
+    register: diff_mapping
+    failed_when: diff_mapping.rc == 1
+
+  - name: Clean up any KubeadmControlPlane node reuse test scenario related temp files.
```

Comment:
This could also work
```
- name: Remove file (delete file)
  ansible.builtin.file:
    path: "{{ item }}"
    state: absent
  with_items:
    - /tmp/before_upgrade_mapping.txt
    - /tmp/after_upgrade_mapping.txt
```